src/main/java/com/exasol/errorreporting/PlaceholderMatcher.java

```diff
@@ -0,0 +1,72 @@
+package com.exasol.errorreporting;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * An iterable for placeholder in a given text.
+ *
+ */
+public class PlaceholderMatcher implements Iterable<Placeholder> {
+    private final String text;
+
+    private PlaceholderMatcher(final String text) {
+        this.text = text;
+    }
+
+    /**
+     * Find placeholders in a given string.
+     * 
+     * @param text string containing placeholders
+     * @return iterable for placeholders
+     */
+    public static PlaceholderMatcher findPlaceholders(final String text) {
```

Comment:
```suggestion
    public static Iterable<Placeholder> findPlaceholders(final String text) {
```

Reply:
perhaps we can even return a `List` instead of an `Iterable` so it's easier to use streams later.

Reply:
added. Returning a list would discard the main advantage of a stream that it directly streams from an iterator and does not need to keep everything in ram. 
Obviously, that's not relevant here, but still....